include/glaze/ext/jsonrpc.hpp

```diff
@@ -0,0 +1,538 @@
+         [](auto const&, auto const&) {}};
+   };
+   namespace concepts
+   {
+      template <typename method_t>
+      concept method_type = requires(method_t meth) {
+                               method_t::name_v;
+                               {
+                                  std::same_as<decltype(method_t::name_v), std::string_view>
+                               };
+                               typename method_t::params_t;
+                               typename method_t::jsonrpc_id_t;
+                               typename method_t::request_t;
+                               typename method_t::response_t;
+                               {
+                                  meth.callback
+                               };
+                            };
+
+      template <typename method_t>
+      concept server_method_type =
+         requires(method_t meth) {
+            requires method_type<method_t>;
+            requires std::same_as<
+               decltype(meth.callback),
+               std::function<expected<typename method_t::result_t, rpc::error>(typename method_t::params_t const&)>>;
+         };
+
+      template <typename method_t>
+      concept client_method_type =
+         requires(method_t meth) {
+            requires method_type<method_t>;
+            requires std::same_as<decltype(meth.callback),
+                                  std::function<void(glz::expected<typename method_t::result_t, rpc::error> const&,
+                                                     jsonrpc_id_type const&)>>;
+         };
+   }
+
+   namespace detail
+   {
+      template <detail::basic_fixed_string name, typename... method_type>
+      inline constexpr void set_callback(glz::tuplet::tuple<method_type...>& methods, auto const& callback)
+      {
+         constexpr bool method_found = ((method_type::name_v == name) || ...);
+         static_assert(method_found, "Method not settable in given tuple.");
+
+         methods.any([&callback](auto&& method) -> bool {
+            if constexpr (name == std::remove_reference_t<decltype(method)>::name_v) {
+               method.callback = callback;
+               return true;
+            }
+            return false;
+         });
+      }
+   }
+
+   template <concepts::server_method_type... method_type>
+   struct server
+   {
+      server() = default;
+      glz::tuplet::tuple<method_type...> methods{};
+
+      template <detail::basic_fixed_string name>
+      constexpr void on(auto const& callback)  // std::function<expected<result_t, rpc::error>(params_t const&)>
+      {
+         detail::set_callback<name>(methods, callback);
+      }
+
+      // Return json stringified response_t meaning it can both be error and non-error
+      // If `id` is null in json_request a response will not be generated
+      std::vector<std::pair<std::string, rpc::error>> call(std::string_view json_request)
+      {
+         if (auto parse_err{glz::validate_json(json_request)}) {
```

Comment:
I would just rely on reading `std::vector<glz::raw_json>` with force_conformance = true for handling parse errors to reduce duplicate parsing.

Reply:
This was done to simplify the error handling since this is the only case where the response should return parse error, in other cases such as params being inconsistent should return the response as invalid request. This handles both `[{},{}]` batch requests and singular requests `{}`.
We should be able to optimize this but I am not sure how it is best implemented.

Reply:
Yeah, that makes sense and optimizations can always wait as long as they don't require api changes, and this shouldnt.